### PR TITLE
Make data the root page and improve auth

### PR DESCRIPTION
**Pull request by ricardocr987**

### CODE REVIEW

File: src/pages/data/DataDetails.tsx [hunk 1]
```diff
@@ -261,7 +261,7 @@ const DataDetails = () => {
             size="lg"
             fullWidth
             icon="home"
-            onClick={() => navigate('/data')}
+            onClick={() => navigate('')}
```
Comment (manomage) [inline src/pages/data/DataDetails.tsx:264]: It should be `navigate('/')`

File: src/shared/components/Prompts.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const PublishedModal: React.FC<{
   publishedName: string;
   isOpen: boolean;
   handleClose: () => void;
-  backTo: '/data' | '/algorithms';
+  backTo: '' | '/algorithms';
```
Comment (manomage) [inline src/shared/components/Prompts.tsx:81]: add the `'/'` here too

File: src/shared/hooks/useSelectData.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import {
   useSearchParams,
 } from 'react-router-dom';
 
-type PathProps = '/data' | '/algorithms' | '/monitor-access' | '/profile';
+type PathProps = '' | '/algorithms' | '/monitor-access' | '/profile';
```
Comment (manomage) [inline src/shared/hooks/useSelectData.ts:7]: add `'/'` here

File: src/components/ui/PriceButton.tsx [hunk 4]
```diff
@@ -85,7 +85,7 @@ export default function PriceButton({
             text="Save"
             onClick={() => {
               handleClose();
-              navigate('/data?tab=published');
+              navigate('?tab=published');
```
[the remark this answers is not in the document]
Comment (ricardocr987) [reply src/components/ui/PriceButton.tsx]: fixed in the new commit

**Review by manomage — APPROVED** [review]

**Pull request merged**